File: JewelryProductionOrder.Models/Gemstone.cs
﻿using System.ComponentModel.DataAnnotations;

namespace JewelryProductionOrder.Models
{
    public class Gemstone
    {
        public int Id { get; set; }
        public string Name { get; set; }
        [Range(0.01, 100000000)]
        public decimal Price { get; set; }
        //[Range(0.01, 100000000)]
        //public decimal Weight { get; set; }
        [Range(0.01, 100000000)]
        public decimal Carat { get; set; }
        public string Color { get; set; }
        public string Clarity { get; set; }
        public string Cut { get; set; }
        public string Status { get; set; }
        public int? MaterialSetId { get; set; }
        public MaterialSet? MaterialSet { get; set; } = null!;
    }
}

File: JewelryProductionOrder/wwwroot/js/gemstone.js


$(document).ready(function () {
    loadDataTable();
})

function loadDataTable() {
    dataTable = $('#tblData').DataTable({
        "ajax": { url: '/gemstone/getall' },
        "columns": [
            { data: 'name' },
            { data: 'price' },
            { data: 'weight' },
            { data: 'carat' },
            { data: 'clarity' },
            { data: 'color' },
            { data: 'cut' },
            {
                data: "id",
                "render": function (data) {
                    return `<div class="w-75 btn-group" role="group">
                    <a href="/gemstone/edit?id=${data}" class="btn btn-primary "> Edit </a>
                    <a href="/gemstone/delete?id=${data}" class="btn btn-danger "> Delete </a>
                    </div>`
                },
            }
        ]
    });
}
